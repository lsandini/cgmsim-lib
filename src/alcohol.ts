import { TreatmentDelta, Treatment, TreatmentDrug, GenderType } from './Types';
import { getDrugActivity } from './drug';
import logger, { getDeltaMinutes, getTreatmentActivity } from './utils';

function getAlcoholActivity(
	gender: 'Male' | 'Female',
	weightKg: number,
	peakMin: number,
	durationMin: number,
	timeMin: number, //minutesAgo
	units: number,
): number {
	const peakAlcoholAmountMin = 60; //min
	const r = gender === 'Male' ? 0.68 : 0.55;
	const eliminationRate = gender === 'Male' ? 0.016 / 60 : 0.018 / 60; //g/100ml/min
	const peakAmount = ((units * 12) / (weightKg * 1000 * r)) * 100; //g/100ml
	const washoutDuration = peakAlcoholAmountMin + peakAmount / eliminationRate; //min
	const unitsWeighted = units * 12 * (80 / weightKg);
	if (washoutDuration < timeMin) {
		return (
			getTreatmentActivity(
				peakMin,
				durationMin,
				timeMin - washoutDuration,
				unitsWeighted,
			) / 35
		);
	}
	return 0;
}

export const computeAlcoholActivity = (
	treatments: TreatmentDelta[],
	weight: number,
	gender: GenderType,
) => {
	const treatmentsActivity = treatments.map((e) => {
		const minutesAgo = e.minutesAgo;
		const units = e.units;
		const activity = getAlcoholActivity(
			gender,
			weight,
			e.peak,
			e.duration,
			minutesAgo,
			units,
		);
		return activity;
	});
	logger.debug('these are the last Alcohol: %o', treatmentsActivity);
	const resultAct = treatmentsActivity.reduce((tot, activity) => {
		return tot + activity;
	}, 0);
	return resultAct;
};

export default function (
	treatments: TreatmentDrug[],
	weight: number,
	gender: GenderType,
): number {
	//Find Alcohol boluses

	const lastALC = getDrugActivity(treatments, weight, 'ALC');
	const activityALC =
		lastALC.length > 0 ? computeAlcoholActivity(lastALC, weight, gender) : 0;
	logger.debug('these are the last ALC: %o', { lastALC, activityALC });

	return activityALC / 2;
}
