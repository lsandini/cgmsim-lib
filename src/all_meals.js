const moment = require('moment');
module.exports = function(treatments = []) {
  const now = moment();
  const meals = treatments
    .filter(e => e.carbs && moment(e.mills).diff(now, 'minutes') > -360)
    .map(e => ({
      ...e,
      time:  (Date.now() - moment(e.mills).valueOf()) / (1000 * 60)
    }));

  console.log('Last 6 hours meal:', meals);
  return meals;
};
