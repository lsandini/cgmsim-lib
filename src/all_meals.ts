import logger from './utils';
import { Treatment, TreatmentDelta } from './Types';
import { getDeltaMinutes } from './utils';

//const logger = pino();

export default function (treatments: Treatment[]): TreatmentDelta[] {

	const meals = treatments
		.filter(e => e.carbs && getDeltaMinutes(e.mills) > 360)
		.map(e => ({
			...e,
			minutesAgo: getDeltaMinutes(e.mills),
		}));

	logger.info('Last 6 hours meal:', meals);
	return meals;
}

