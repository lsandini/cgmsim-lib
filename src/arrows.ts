import logger from './utils';
import { Sgv, Direction } from './Types';

//const logger = pino();


export default function (sgvs: Pick<Sgv, 'sgv'>[]): { sgvdir: number; direction: Direction }[] {
	const arrows = [];
	if (sgvs && sgvs.length >= 4) {
		const sgvdir1 = sgvs[0].sgv - sgvs[1].sgv;
		const sgvdir2 = sgvs[1].sgv - sgvs[2].sgv;
		const sgvdir3 = sgvs[2].sgv - sgvs[3].sgv;
		const sgvdir15min = (sgvdir1 + sgvdir2 + sgvdir3) / 3;
		logger.info('this is the mean SGV 5 min variation in the last 15 minutes: %o', sgvdir15min, 'mg/dl');

		if (sgvdir15min < -10) {
			arrows.push({
				sgvdir: sgvdir15min,
				direction: 'DoubleDown'
			});

		} else if (sgvdir15min < -6) {
			arrows.push({
				sgvdir: sgvdir15min,
				direction: 'SingleDown'
			});

		} else if (sgvdir15min < -2) {
			arrows.push({
				sgvdir: sgvdir15min,
				direction: 'FortyFiveDown'
			});

		} else if (sgvdir15min < 2) {
			arrows.push({
				sgvdir: sgvdir15min,
				direction: 'Flat'
			});

		} else if (sgvdir15min < 6) {
			arrows.push({
				sgvdir: sgvdir15min,
				direction: 'FortyFiveUp'
			});

		} else if (sgvdir15min < 10) {
			arrows.push({
				sgvdir: sgvdir15min,
				direction: 'SingleUp'
			});

		} else if (sgvdir15min >= 10) {
			arrows.push({
				sgvdir: sgvdir15min,
				direction: 'DoubleUp'
			});
		}

	} else {
		arrows.push({
			sgvdir: 0,
			direction: 'Flat'
		});
	}
	logger.info(arrows);

	return arrows;
}