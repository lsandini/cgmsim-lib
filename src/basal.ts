import { TreatmentDelta, Treatment } from './Types';
import logger, { getDeltaMinutes, getInsulinActivity } from './utils';

export const peakBasal = {
	GLA: (duration: number) => duration / 2.5,
	DET: (duration: number) => duration / 3,
	TOU: (duration: number) => duration / 2.5,
	DEG: (duration: number) => duration / 3,
};
export const durationBasal = {
	GLA: (insulin: number, weight: number) => (22 + (12 * insulin) / weight) * 60,
	DET: (insulin: number, weight: number) => (14 + (24 * insulin) / weight) * 60,
	TOU: (insulin: number, weight: number) => (24 + (14 * insulin) / weight) * 60,
	DEG: () => 42 * 60,
};

export const computeBasalActivity = (
	treatments: (TreatmentDelta & { duration: number; peak: number })[]
) => {
	// activities be expressed as U/min !!!
	const treatmentsActivity = treatments.map((e) => {
		const minutesAgo = e.minutesAgo;
		const insulin = e.insulin;
		const activity = getInsulinActivity(
			e.peak,
			e.duration,
			minutesAgo,
			insulin
		);
		return activity;
	});
	logger.debug('these are the last Slow INSULINS: %o', treatmentsActivity);
	const resultAct = treatmentsActivity.reduce((tot, activity) => {
		return tot + activity;
	}, 0);
	return resultAct;
};

export default function (treatments: Treatment[], weight: number): number {
	//Find basal boluses
	const basals =
		treatments && treatments.length
			? treatments
					.filter((e) => e.notes)
					.map((e) => {
						const lastIndexEmptySpace = e.notes.lastIndexOf(' ');
						logger.debug(
							'tou %o',
							parseInt(e.notes.slice(lastIndexEmptySpace), 10)
						);
						return {
							...e,
							minutesAgo: getDeltaMinutes(e.created_at),
							drug: e.notes.slice(0, 3),
							// insulin: parseInt(e.notes.slice(-2))
							insulin: parseInt(e.notes.slice(lastIndexEmptySpace), 10) || 0,
						};
					})
					.filter((e) => e.minutesAgo >= 0)
			: [];

	const lastBasals = basals.filter(function (e) {
		return e.minutesAgo <= 45 * 60; // keep only the basals from the last 45 hours
	});

	const lastGLA = lastBasals
		.filter((e) => {
			return (
				e.drug === 'gla' ||
				e.drug === 'Gla' ||
				e.drug === 'lan' ||
				e.drug === 'Lan'
			); // keep only the glas from the last 45 hours
		})
		.map((e) => {
			const duration = durationBasal.GLA(e.insulin, weight);
			const peak = peakBasal.GLA(duration);
			return {
				...e,
				duration,
				peak,
			};
		});
	const activityGLA = lastGLA.length > 0 ? computeBasalActivity(lastGLA) : 0;
	logger.debug('these are the last GLA: %o', { lastGLA, activityGLA });

	const lastDET = lastBasals
		.filter(function (e) {
			return (
				e.drug === 'det' ||
				e.drug === 'Det' ||
				e.drug === 'lev' ||
				e.drug === 'Lev'
			); // keep only the dets from the last 45 hours
		})
		.map((e) => {
			const duration = durationBasal.DET(e.insulin, weight);
			const peak = peakBasal.DET(duration);
			return {
				...e,
				duration,
				peak,
			};
		});

	const activityDET = lastDET.length ? computeBasalActivity(lastDET) : 0;
	logger.debug('these are the last DET: %o', { lastDET, activityDET });

	const lastTOU = lastBasals
		.filter(function (e) {
			return e.drug === 'tou' || e.drug === 'Tou'; // keep only the tous from the last 45 hours
		})
		.map((e) => {
			const duration = durationBasal.TOU(e.insulin, weight);
			const peak = peakBasal.TOU(duration);
			return {
				...e,
				duration,
				peak,
			};
		});
	const activityTOU = lastTOU.length ? computeBasalActivity(lastTOU) : 0;
	logger.debug('these are the last TOU: %o', { lastTOU, activityTOU });

	const lastDEG = lastBasals
		.filter(function (e) {
			return (
				e.drug === 'deg' ||
				e.drug === 'Deg' ||
				e.drug === 'tre' ||
				e.drug === 'Tre'
			); // keep only the degs from the last 45 hours
		})
		.map((e) => {
			const duration = durationBasal.DEG();
			const peak = peakBasal.DEG(duration);
			return {
				...e,
				duration,
				peak,
			};
		});
	const activityDEG = lastDEG.length ? computeBasalActivity(lastDEG) : 0;
	logger.debug('these are the last deg: %o', { lastDEG, activityDEG });


	return activityDEG + activityDET + activityGLA + activityTOU ;
}
