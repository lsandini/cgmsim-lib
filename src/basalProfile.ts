import moment = require('moment');
import { Profile } from './Types';
export default function (profile: Profile[]): number {
	let lastProfile: Profile = null;
	let lastProfiles = profile.sort((first, second) =>
		moment(second.startDate).diff(-moment(first.startDate)),
	);

	if (!lastProfiles || lastProfiles.length === 0) {
		return 0;
	}
	lastProfile = lastProfiles[0];
	const defaultProfile = lastProfile.defaultProfile;
	const defaultProfileBasals = Array.isArray(
		lastProfile.store[defaultProfile].basal,
	)
		? lastProfile.store[defaultProfile].basal[0].value
		: lastProfile.store[defaultProfile].basal;
	return defaultProfileBasals / 60;
}
