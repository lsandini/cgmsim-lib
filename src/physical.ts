/**
 * @fileoverview Physical activity calculations and analysis module
 * Handles heart rate and steps data to determine activity intensity,
 * insulin sensitivity, and liver glucose production factors.
 */

// Import necessary functions and types
import { getMaxHr, physicalHeartIntensity, physicalHeartRateIsf, physicalHeartRateLiver } from './heartRate';
import { physicalStepsIsf, physicalStepsLiver } from './steps';
import { Activity, GenderType } from './Types';
import logger, { getDeltaMinutes } from './utils';

// Constants for activity calculations
const ACTIVITY_CONSTANTS = {
	DEFAULT_INTENSITY: 0,
	TIME_WINDOW_MINUTES: 360, // 6 hours
	MIN_ACTIVITY_THRESHOLD: 0,
} as const;

// Type definitions for activities with time information
interface ActivityWithTime extends Activity {
	minutesAgo: number;
}

/**
 * Adds time information to each activity
 * @param activities - Array of raw activity data
 * @returns Array of activities with added time information
 */
function addTimeToActivities(activities: Activity[]): ActivityWithTime[] {
	return activities.map((activity) => ({
		...activity,
		minutesAgo: getDeltaMinutes(activity.created_at),
	}));
}

/**
 * Calculates the current intensity of physical activity based on user parameters.
 * This function combines heart rate data to determine activity intensity level.
 * @param activities - Array of physical activity data containing heart rate and timestamp
 * @param age - Patient's age
 * @param gender - Patient's gender
 * @returns Current intensity of physical activity as a percentage (0-100)
 */
export function currentIntensity(activities: Activity[], age: number, gender: GenderType): number {
	if (!activities?.length) {
		return ACTIVITY_CONSTANTS.DEFAULT_INTENSITY;
	}

	const maxHeartRate = getMaxHr(age, gender);
	const activitiesWithTime = addTimeToActivities(activities);

	const intensity = physicalHeartIntensity(activitiesWithTime, maxHeartRate);
	return intensity ?? ACTIVITY_CONSTANTS.DEFAULT_INTENSITY;
}

/**
 * Calculates insulin sensitivity factor based on physical activity.
 * Compares heart rate and steps data to determine the most significant impact.
 * @param activities - Array of physical activities with heart rate and steps
 * @param age - Patient's age
 * @param gender - Patient's gender
 * @returns The higher insulin sensitivity factor between heart rate and steps calculations
 */
export function physicalIsf(activities: Activity[], age: number, gender: GenderType): number {
	const maxHeartRate = getMaxHr(age, gender);
	const activitiesWithTime = addTimeToActivities(activities);

	const heartRateIsf = physicalHeartRateIsf(activitiesWithTime, maxHeartRate);
	const stepsIsf = physicalStepsIsf(activitiesWithTime);

	if (heartRateIsf > stepsIsf) {
		logger.info('[Physical Activity] Heart rate ISF calculation used: %d', heartRateIsf);
	} else {
		logger.info('[Physical Activity] Steps ISF calculation used: %d', stepsIsf);
	}

	return 1 + Math.max(heartRateIsf, stepsIsf);
}

/**
 * Calculates liver glucose production factor based on physical activity.
 * Compares heart rate and steps data to determine the strongest impact on liver glucose production.
 * @param activities - Array of physical activities with heart rate and steps
 * @param age - Patient's age
 * @param gender - Patient's gender
 * @returns The higher liver glucose production factor between heart rate and steps calculations
 */
export function physicalLiver(activities: Activity[], age: number, gender: GenderType): number {
	const maxHeartRate = getMaxHr(age, gender);
	const activitiesWithTime = addTimeToActivities(activities);

	return 1 + Math.max(physicalHeartRateLiver(activitiesWithTime, maxHeartRate), physicalStepsLiver(activitiesWithTime));
}
