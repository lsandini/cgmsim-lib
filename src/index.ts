import simulator from './CGMSIMsimulator';
import simulatorUVA from './UVAsimulator';
import downloads from './downloads';
import uploadEntries from './upload-entries';
import uploadActivity from './upload-activity';
import loadActivity from './load-activity';
import uploadNotes from './upload-notes';
import arrows from './arrows';


export { arrows, simulator, downloads, uploadEntries, uploadActivity, loadActivity, simulatorUVA, uploadNotes };
