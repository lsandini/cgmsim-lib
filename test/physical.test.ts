import moment = require('moment');
import { physicalIsf } from '../src/physical';
import { physicalLiver } from '../src/physical';
import { getFlatHeartRate } from './inputTest';
describe('Physical test', () => {


	beforeAll(() => {
		jest.useFakeTimers('modern');
		jest.setSystemTime(new Date('2001-01-01'));
	});

	afterAll(() => {
		jest.useRealTimers();
	});
	// test('ISF: physicalIsf with flat hr =0.5', () => {
	// 	const activities = getFlatHeartRate({ heartRate: 170 * 0.5, created_at: '2001-01-01T00:00:00.000Z' }, 6);
	// 	const result = physicalIsf(activities, 51, 'Male');
	// 	jest.setSystemTime(new Date('2001-01-01T06:00:00.000Z'));
	// 	expect(result).toBe(1);
	// })

	// test('test flat 5hr=0.5 1h=0.7', () => {
	// 	jest.setSystemTime(new Date('2001-01-01T06:00:00.000Z'));
	// 	const activities05 = getFlatHeartRate({ heartRate: 170 * 0.5, created_at: '2001-01-01T00:00:00.000Z' }, 5);
	// 	const activities07 = getFlatHeartRate({ heartRate: 170 * 0.7, created_at: '2001-01-01T05:00:00.000Z' }, 1);
	// 	const result = physicalIsf([...activities05, ...activities07]);
	// 	expect(result).toBe(2.076371170507074);
	// })

	describe('ISF: physicalISF with 6h0.5 + 2h0.91 + 6h0.5', () => {
		const activities05 = getFlatHeartRate({ heartRate: 170 * 0.5, created_at: '2001-01-01T00:00:00.000Z' }, 6);
		const activities07 = getFlatHeartRate({ heartRate: 170 * 0.91, created_at: '2001-01-01T06:00:00.000Z' }, 2);
		const activities05n = getFlatHeartRate({ heartRate: 170 * 0.5, created_at: '2001-01-01T08:00:00.000Z' }, 6);

		test.each([...activities05, ...activities07, ...activities05n])('%p', (t) => {
			jest.setSystemTime(new Date(t.created_at));
			const pastActivities = [...activities05, ...activities07,...activities05n].filter(a=>a.created_at<=t.created_at)
			const result = physicalIsf(pastActivities, 51, 'Male');
			expect(result).toMatchSnapshot();
		})
	});

	describe('Liver: physicalLiver with 6h0.5 + 2h0.91 + 6h0.5', () => {
		const activities05 = getFlatHeartRate({ heartRate: 170 * 0.5, created_at: '2001-01-01T00:00:00.000Z' }, 6);
		const activities07 = getFlatHeartRate({ heartRate: 170 * 0.91, created_at: '2001-01-01T06:00:00.000Z' }, 2);
		const activities05n = getFlatHeartRate({ heartRate: 170 * 0.5, created_at: '2001-01-01T08:00:00.000Z' }, 6);

		test.each([...activities05, ...activities07, ...activities05n])('%p', (t) => {
			jest.setSystemTime(new Date(t.created_at));
			const pastActivities = [...activities05, ...activities07,...activities05n].filter(a=>a.created_at<=t.created_at)
			const result = physicalLiver(pastActivities, 51, 'Male');
			expect(result).toMatchSnapshot();
		})
	});


})