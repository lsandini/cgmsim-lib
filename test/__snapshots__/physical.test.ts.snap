// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Physical test test sequence {"created_at": "2001-01-01T00:00:00.000Z", "heartRate": 85} 1`] = `1`;

exports[`Physical test test sequence {"created_at": "2001-01-01T00:05:00.000Z", "heartRate": 85} 1`] = `1`;

exports[`Physical test test sequence {"created_at": "2001-01-01T00:10:00.000Z", "heartRate": 85} 1`] = `1`;

exports[`Physical test test sequence {"created_at": "2001-01-01T00:15:00.000Z", "heartRate": 85} 1`] = `1`;

exports[`Physical test test sequence {"created_at": "2001-01-01T00:20:00.000Z", "heartRate": 85} 1`] = `1`;

exports[`Physical test test sequence {"created_at": "2001-01-01T00:25:00.000Z", "heartRate": 85} 1`] = `1`;

exports[`Physical test test sequence {"created_at": "2001-01-01T00:30:00.000Z", "heartRate": 85} 1`] = `1`;

exports[`Physical test test sequence {"created_at": "2001-01-01T00:35:00.000Z", "heartRate": 85} 1`] = `1`;

exports[`Physical test test sequence {"created_at": "2001-01-01T00:40:00.000Z", "heartRate": 85} 1`] = `1`;

exports[`Physical test test sequence {"created_at": "2001-01-01T00:45:00.000Z", "heartRate": 85} 1`] = `1`;

exports[`Physical test test sequence {"created_at": "2001-01-01T00:50:00.000Z", "heartRate": 85} 1`] = `1`;

exports[`Physical test test sequence {"created_at": "2001-01-01T00:55:00.000Z", "heartRate": 85} 1`] = `1`;

exports[`Physical test test sequence {"created_at": "2001-01-01T01:00:00.000Z", "heartRate": 85} 1`] = `1`;

exports[`Physical test test sequence {"created_at": "2001-01-01T01:05:00.000Z", "heartRate": 85} 1`] = `1`;

exports[`Physical test test sequence {"created_at": "2001-01-01T01:10:00.000Z", "heartRate": 85} 1`] = `1`;

exports[`Physical test test sequence {"created_at": "2001-01-01T01:15:00.000Z", "heartRate": 85} 1`] = `1`;

exports[`Physical test test sequence {"created_at": "2001-01-01T01:20:00.000Z", "heartRate": 85} 1`] = `1`;

exports[`Physical test test sequence {"created_at": "2001-01-01T01:25:00.000Z", "heartRate": 85} 1`] = `1`;

exports[`Physical test test sequence {"created_at": "2001-01-01T01:30:00.000Z", "heartRate": 85} 1`] = `1`;

exports[`Physical test test sequence {"created_at": "2001-01-01T01:35:00.000Z", "heartRate": 85} 1`] = `1`;

exports[`Physical test test sequence {"created_at": "2001-01-01T01:40:00.000Z", "heartRate": 85} 1`] = `1`;

exports[`Physical test test sequence {"created_at": "2001-01-01T01:45:00.000Z", "heartRate": 85} 1`] = `1`;

exports[`Physical test test sequence {"created_at": "2001-01-01T01:50:00.000Z", "heartRate": 85} 1`] = `1`;

exports[`Physical test test sequence {"created_at": "2001-01-01T01:55:00.000Z", "heartRate": 85} 1`] = `1`;

exports[`Physical test test sequence {"created_at": "2001-01-01T02:00:00.000Z", "heartRate": 85} 1`] = `1`;

exports[`Physical test test sequence {"created_at": "2001-01-01T02:05:00.000Z", "heartRate": 85} 1`] = `1`;

exports[`Physical test test sequence {"created_at": "2001-01-01T02:10:00.000Z", "heartRate": 85} 1`] = `1`;

exports[`Physical test test sequence {"created_at": "2001-01-01T02:15:00.000Z", "heartRate": 85} 1`] = `1`;

exports[`Physical test test sequence {"created_at": "2001-01-01T02:20:00.000Z", "heartRate": 85} 1`] = `1`;

exports[`Physical test test sequence {"created_at": "2001-01-01T02:25:00.000Z", "heartRate": 85} 1`] = `1`;

exports[`Physical test test sequence {"created_at": "2001-01-01T02:30:00.000Z", "heartRate": 85} 1`] = `1`;

exports[`Physical test test sequence {"created_at": "2001-01-01T02:35:00.000Z", "heartRate": 85} 1`] = `1`;

exports[`Physical test test sequence {"created_at": "2001-01-01T02:40:00.000Z", "heartRate": 85} 1`] = `1`;

exports[`Physical test test sequence {"created_at": "2001-01-01T02:45:00.000Z", "heartRate": 85} 1`] = `1`;

exports[`Physical test test sequence {"created_at": "2001-01-01T02:50:00.000Z", "heartRate": 85} 1`] = `1`;

exports[`Physical test test sequence {"created_at": "2001-01-01T02:55:00.000Z", "heartRate": 85} 1`] = `1`;

exports[`Physical test test sequence {"created_at": "2001-01-01T03:00:00.000Z", "heartRate": 85} 1`] = `1`;

exports[`Physical test test sequence {"created_at": "2001-01-01T03:05:00.000Z", "heartRate": 85} 1`] = `1`;

exports[`Physical test test sequence {"created_at": "2001-01-01T03:10:00.000Z", "heartRate": 85} 1`] = `1`;

exports[`Physical test test sequence {"created_at": "2001-01-01T03:15:00.000Z", "heartRate": 85} 1`] = `1`;

exports[`Physical test test sequence {"created_at": "2001-01-01T03:20:00.000Z", "heartRate": 85} 1`] = `1`;

exports[`Physical test test sequence {"created_at": "2001-01-01T03:25:00.000Z", "heartRate": 85} 1`] = `1`;

exports[`Physical test test sequence {"created_at": "2001-01-01T03:30:00.000Z", "heartRate": 85} 1`] = `1`;

exports[`Physical test test sequence {"created_at": "2001-01-01T03:35:00.000Z", "heartRate": 85} 1`] = `1`;

exports[`Physical test test sequence {"created_at": "2001-01-01T03:40:00.000Z", "heartRate": 85} 1`] = `1`;

exports[`Physical test test sequence {"created_at": "2001-01-01T03:45:00.000Z", "heartRate": 85} 1`] = `1`;

exports[`Physical test test sequence {"created_at": "2001-01-01T03:50:00.000Z", "heartRate": 85} 1`] = `1`;

exports[`Physical test test sequence {"created_at": "2001-01-01T03:55:00.000Z", "heartRate": 85} 1`] = `1`;

exports[`Physical test test sequence {"created_at": "2001-01-01T04:00:00.000Z", "heartRate": 85} 1`] = `1`;

exports[`Physical test test sequence {"created_at": "2001-01-01T04:05:00.000Z", "heartRate": 85} 1`] = `1`;

exports[`Physical test test sequence {"created_at": "2001-01-01T04:10:00.000Z", "heartRate": 85} 1`] = `1`;

exports[`Physical test test sequence {"created_at": "2001-01-01T04:15:00.000Z", "heartRate": 85} 1`] = `1`;

exports[`Physical test test sequence {"created_at": "2001-01-01T04:20:00.000Z", "heartRate": 85} 1`] = `1`;

exports[`Physical test test sequence {"created_at": "2001-01-01T04:25:00.000Z", "heartRate": 85} 1`] = `1`;

exports[`Physical test test sequence {"created_at": "2001-01-01T04:30:00.000Z", "heartRate": 85} 1`] = `1`;

exports[`Physical test test sequence {"created_at": "2001-01-01T04:35:00.000Z", "heartRate": 85} 1`] = `1`;

exports[`Physical test test sequence {"created_at": "2001-01-01T04:40:00.000Z", "heartRate": 85} 1`] = `1`;

exports[`Physical test test sequence {"created_at": "2001-01-01T04:45:00.000Z", "heartRate": 85} 1`] = `1`;

exports[`Physical test test sequence {"created_at": "2001-01-01T04:50:00.000Z", "heartRate": 85} 1`] = `1`;

exports[`Physical test test sequence {"created_at": "2001-01-01T04:55:00.000Z", "heartRate": 85} 1`] = `1`;

exports[`Physical test test sequence {"created_at": "2001-01-01T05:00:00.000Z", "heartRate": 118.99999999999999} 1`] = `1.7`;

exports[`Physical test test sequence {"created_at": "2001-01-01T05:05:00.000Z", "heartRate": 118.99999999999999} 1`] = `2.124571461798843`;

exports[`Physical test test sequence {"created_at": "2001-01-01T05:10:00.000Z", "heartRate": 118.99999999999999} 1`] = `2.382087070618853`;

exports[`Physical test test sequence {"created_at": "2001-01-01T05:15:00.000Z", "heartRate": 118.99999999999999} 1`] = `2.538278182722754`;

exports[`Physical test test sequence {"created_at": "2001-01-01T05:20:00.000Z", "heartRate": 118.99999999999999} 1`] = `2.6330128809883826`;

exports[`Physical test test sequence {"created_at": "2001-01-01T05:25:00.000Z", "heartRate": 118.99999999999999} 1`] = `2.6904723800251116`;

exports[`Physical test test sequence {"created_at": "2001-01-01T05:30:00.000Z", "heartRate": 118.99999999999999} 1`] = `2.7253233278826166`;

exports[`Physical test test sequence {"created_at": "2001-01-01T05:35:00.000Z", "heartRate": 118.99999999999999} 1`] = `2.74646149627824`;

exports[`Physical test test sequence {"created_at": "2001-01-01T05:40:00.000Z", "heartRate": 118.99999999999999} 1`] = `2.7592824435003536`;

exports[`Physical test test sequence {"created_at": "2001-01-01T05:45:00.000Z", "heartRate": 118.99999999999999} 1`] = `2.7670587410771232`;

exports[`Physical test test sequence {"created_at": "2001-01-01T05:50:00.000Z", "heartRate": 118.99999999999999} 1`] = `2.771775303976483`;

exports[`Physical test test sequence {"created_at": "2001-01-01T05:55:00.000Z", "heartRate": 118.99999999999999} 1`] = `2.774636043983408`;
